File: apier-auth/src/main/java/cn/apier/auth/domain/model/ClientApplication.java
package cn.apier.auth.domain.model;

import cn.apier.common.util.DateTimeUtil;

import javax.persistence.Entity;
import javax.persistence.Id;
import java.util.Date;

/**
 * Created by yanjunhua on 2017/5/23.
 * 客户端应用
 */
@Entity
public class ClientApplication
{
    @Id
    private String uid;
    private String name;
    private String appKey;
    private String secretKey;
    private Date createdAt = DateTimeUtil.now();

    private ClientApplication()
    {
    }

    public ClientApplication(String uid, String name, String appKey, String secretKey)
    {
        this.uid = uid;
        this.name = name;
        this.appKey = appKey;
        this.secretKey = secretKey;
    }

    public String getUid()
    {
        return uid;
    }

    public String getName()
    {
        return name;
    }

    public String getAppKey()
    {
        return appKey;
    }

    public String getSecretKey()
    {
        return secretKey;
    }

    public Date getCreatedAt()
    {
        return createdAt;
    }
}

File: apier-auth/src/main/kotlin/cn/apier/auth/domain/model/ClientApplication.kt
package cn.apier.auth.domain.model

import cn.apier.auth.domain.event.ClientApplicationCreatedEvent
import cn.apier.common.domain.model.EnterpriseDeletableBaseModel
import org.axonframework.commandhandling.model.AggregateLifecycle
import org.axonframework.eventsourcing.EventSourcingHandler
import org.axonframework.spring.stereotype.Aggregate
import java.util.*

@Aggregate
class ClientApplication : EnterpriseDeletableBaseModel {

    private var name: String = ""
    private var appKey: String = ""
    private var appSecret: String = ""

    private constructor()
    constructor(enterpriseId: String, uid: String, name: String, appKey: String, appSecret: String, deleted: Boolean, createdAt: Date) {
        AggregateLifecycle.apply(ClientApplicationCreatedEvent(enterpriseId, uid, name, appKey, appSecret, deleted, createdAt))
    }


    @EventSourcingHandler
    private fun onCreated(clientApplicationCreatedEvent: ClientApplicationCreatedEvent) {
        this.enterpriseId = clientApplicationCreatedEvent.enterpriseId
        this.uid = clientApplicationCreatedEvent.uid
        this.deleted = clientApplicationCreatedEvent.deleted
        this.appKey = clientApplicationCreatedEvent.appKey
        this.appSecret = clientApplicationCreatedEvent.appSecret
        this.createdAt = clientApplicationCreatedEvent.createdAt
        this.name = clientApplicationCreatedEvent.name
    }
}